tests/base_tests.py

```diff
@@ -529,6 +553,48 @@ def test_original_df_not_updated(self, initialized_transformers):
         pd.testing.assert_frame_equal(df, d.create_df_3())
 
 
+class DropOriginalTransformTests(GenericTransformTests):
+    """
+    Transform tests for transformers that take a "drop_original" argument
+    Note this deliberately avoids starting with "Tests" so that the tests are not run on import.
+    """
+
+    def test_original_columns_dropped_when_specified(
+        self,
+        initialized_transformers,
+    ):
+        """Test transformer drops original columns when specified."""
+
+        df = d.create_df_3()
+
+        x = initialized_transformers[self.transformer_name]
+
+        x.columns = ["a", "b"]
+        x.drop_original = True
+
+        df_transformed = x.transform(df)
+
+        assert ("a" not in df_transformed.columns.to_numpy()) and (
+            "b" not in df_transformed.columns.to_numpy()
+        ), "original columns not dropped"
+
+    def test_original_columns_kept_when_specified(self, initialized_transformers):
+        """Test transformer keeps original columns when specified."""
+
+        df = d.create_df_3()
+
+        x = initialized_transformers[self.transformer_name]
+
+        x.columns = ["a", "b"]
+        x.drop_original = False
+
+        df_transformed = x.transform(df)
+
+        assert ("a" in df_transformed.columns.to_numpy()) and (
+            "b" in df_transformed.columns.to_numpy()
+        ), "original columns not kept"
+
```

Comment:
S - ideally would also have a test that other columns in X  are unaffected when dropping columns addressed by the transformer.

Reply:
Thanks for reviewing David, have pushed recommend changes :)